### ISSUE TITLE
mAp only 2% after train 2 days on coco2017

### ISSUE THREAD

**Issue by sunny123123** [eriklindernoren/PyTorch-YOLOv3#683]
+ problem
Problem description same as title, I got low mAP on coco2017 datasets. Only 2% mAP when had trained two days.
![image](https://user-images.githubusercontent.com/10163641/117652011-3a511c80-b1c5-11eb-9ffb-5a2fa6d39780.png)

+ code version
![image](https://user-images.githubusercontent.com/10163641/117651893-14c41300-b1c5-11eb-8785-f60d2ba55e6c.png)

+ train command
just use default command
```
python pytorchyolo/train.py 
```
+ train data format 
same as get_coco_dataset.sh. I sure data format is right
```
49 0.4729499999999999 0.52 0.66706 0.7153066666666666
0 0.09570999999999999 0.5420666666666667 0.18797999999999998 0.74936
``` 

I think it's not right, please give me some suggestions.

**Comment by sunny123123** [comment]
@Flova  oh, thanks for you relay quickly. I train indeed with nothing pre-trained weights. so far, I have trained 69 epochs on one GPU card **RTX 2080 ti** with batch size 10. mAP has improved a little bit to 2.24%, but it's not enough.
![image](https://user-images.githubusercontent.com/10163641/117748847-55627180-b243-11eb-870c-99a06e8d3471.png)

so , I decide to follow your advice. I will try training with **pre-trained weights darknet53.conv.74**. But I want to make sure of two things with you.
(1) Metrics with mAP in TensorBoard is mAP[iou=0.5], not mAP[IoU=0.50:0.95], right?
(2)In your code, you didn't execute **detach() op** after load pre-trained  weights `darknet53.conv.74`. why? it will update darknet weights if you not detach from graph. 

Thanks again for you relay.

**Comment by Flova** [comment]
Why did you train with a batch size of 10? The default is 16. Changing the batch size influences the stability of the training. You can subdivide a batch by increasing `subdivisions` if your batch doesn't fit on your GPU.

69 Epochs are still way too few. Even if you use pre-trained weights.

Regarding your questions:

* Yes indeed. We use mAP50
* Yes the pre-trained weights are used as an initialization. They are still updated during the training. If you want to freeze certain layers try setting `requires_grad=False` for these layers after loading the model. (https://stackoverflow.com/questions/51748138/pytorch-how-to-set-requires-grad-false) I wouldn't use detach for this purpose. Or is there a misunderstanding on my side?

Cheers

**Comment by sunny123123** [comment]
@Flova oh, thanks. My GPU only support batch size 10. Indeed, it's very slowly though I used pre-trained weights.

![image](https://user-images.githubusercontent.com/10163641/117929163-0b56ba00-b32f-11eb-90c2-8707f16f4b8b.png)
+ **red corve**: without darknet pre-trained weights
+ **blue corve**: with darknet pre-trained weights

**Comment by Flova** [comment]
It looks like your training speed increases drastically after the default learning rate reduction. So maybe try and lower it by factor 10.
